src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/InliningInfoNode.cs

```diff
@@ -66,6 +66,13 @@ public override ObjectData GetData(NodeFactory factory, bool relocsOnly = false)
                         continue;
                     }
 
+                    if (!factory.CompilationModuleGroup.VersionsWithMethodBody(inlinee))
+                    {
+                        // We cannot record inlining info across version bubble as cross-bubble assemblies
```

Comment:
We should assert that this only happens for NonVersionable methods. (If its happening for any other method, we shouldn't actually be inclining)

Reply:
Good point, fixed in 3rd commit. Could you please take another look when you have a chance?